components/CDT/wmtk/components/CDT/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+set(COMPONENT_NAME CDT)
+add_component(${COMPONENT_NAME})
+if(NOT ${WMTK_ENABLE_COMPONENT_${COMPONENT_NAME}})
+    return()
+endif()
+
+set(SRC_FILES
+    internal/CDTOptions.hpp
+    internal/CDT.hpp
+    internal/CDT.cpp
+    CDT.hpp
+    CDT.cpp
+    )
+
+include(CDT)
+
+
+target_sources(wmtk_${COMPONENT_NAME} PRIVATE ${SRC_FILES})
+
+add_library(wmtk_cdt_get_vf 
+    internal/get_vf.hpp
+    internal/get_vf.cpp
+)
+
+add_library(wmtk_cdt_lib
```

Comment:
why?

Reply:
Eigen cannot work with marco's code in the same library. It somehow messed up Eigen's destructor.